spidermon/contrib/validation/jsonschema/validator.py

```diff
@@ -18,22 +17,19 @@ class JSONSchemaValidator(Validator):
 
     def __init__(self, schema, translator=None, use_default_translator=True):
         super(JSONSchemaValidator, self).__init__(
-            translator=translator,
-            use_default_translator=use_default_translator,
+            translator=translator, use_default_translator=use_default_translator
         )
         self._schema = schema
 
     def _validate(self, data, strict=False):
-        validator = Draft4Validator(
-            schema=self._schema,
-            format_checker=format_checker,
-        )
+        validator_cls = validator_for(self._schema)
+        validator = validator_cls(schema=self._schema, format_checker=format_checker)
         errors = validator.iter_errors(data)
-        for e in errors:
-            #print e
-            absolute_path = list(e.absolute_path)
-            required_match = REQUIRED_RE.search(e.message)
+
+        for error in errors:
+            absolute_path = list(error.absolute_path)
+            required_match = REQUIRED_RE.search(error.message)
             if required_match:
                 absolute_path.append(required_match.group(1))
-            field_name = '.'.join([str(p) for p in absolute_path])
-            self._add_errors({field_name: [e.message]})
+            field_name = '.'.join([p for p in absolute_path])
```

Comment:
@rennerocha Don't believe me, [absolute_path](https://github.com/Julian/jsonschema/blob/6f10ce6ce2379df94a17ea41c671c71b107ea99f/jsonschema/exceptions.py#L99) can contain `int`

Reply:
I noticed that... :-)